lib/nmatrix/blas.rb

```diff
@@ -73,18 +72,19 @@ class << self
     #   - +ArgumentError+ -> The dtype of the matrices must be equal.
     #
     def gemm(a, b, c = nil, alpha = 1.0, beta = 0.0,
-             transpose_a = false, transpose_b = false, m = nil,
-             n = nil, k = nil, lda = nil, ldb = nil, ldc = nil)
+      transpose_a = false, transpose_b = false, m = nil,
+      n = nil, k = nil, lda = nil, ldb = nil, ldc = nil)
 
-      raise(ArgumentError, 'Expected dense NMatrices as first two arguments.') \
-            unless a.is_a?(NMatrix) and b.is_a? \
-            (NMatrix) and a.stype == :dense and b.stype == :dense
+      raise(ArgumentError, "Expected dense NMatrices as first two arguments.") \
```

Comment:
I think you could separate these validations into multiple functions to make this function more readable:
```rb
def gemm(.....)
  ...
  validate_argument?(a)
  validate_argument?(b)
  ...
end

private

def validate_argument(matrix)
  nmatrix?(matrix) && dense_stype?(matrix)
end

def nmatrix?(matrix)
  message = 'Expected NMatrices object as first two arguments.'
  raise(ArgumentError, message) unless matrix.is_a?(NMatrix)
end

def dense_stype?(matrix)
  message = 'Expected dense NMatrices as first two arguments.'
  raise(ArgumentError, message) unless (matrix.stype == :dense) 
end
```
WDYT? 🤔